lib/API/JsonApi/v1/Filters/Parsers/FilterParser.php

```diff
@@ -341,10 +341,9 @@ protected function isInOperatorsArray($operatorNeedle, array $operatorsHaystack)
      * @return string
      */
     private function stringDifference($a, $b) {
-        $aArray = str_split($a);
-        $bArray = str_split($b);
-        $arrayDiff = array_diff($aArray, $bArray);
-        return implode('', $arrayDiff);
+        if (substr($a, 0, strlen($b)) == $b) {
```

Comment:
`stringDifference('php', 'h')` doesn't return anything?

Reply:
I will add the empty string return.

According to you, what should the function do?

Reply:
Well, the way I interpret the name is that this function will return the difference between two strings. But then the original body seems to do just that, so I might be wrong. I haven't looked further into this issue, just reviewing this particular method.

Another thing you want to watch out for is the comparison operator, you want strict because otherwise ('01' == '1') equals true, and those are not equal strings.